FhirToDataLake/test/Microsoft.Health.Fhir.Synapse.E2ETests/DicomToDataLakeE2ETests.cs

```diff
@@ -0,0 +1,272 @@
+        private AzureStorageClientFactory _queueClientFactory;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DicomToDataLakeE2ETests"/> class.
+        /// </summary>
+        /// <param name="testOutputHelper">output helper.</param>
+        public DicomToDataLakeE2ETests(ITestOutputHelper testOutputHelper)
+        {
+            _testOutputHelper = testOutputHelper;
+
+            string storageUri = Environment.GetEnvironmentVariable("dataLakeStore:storageUrl");
+            if (!string.IsNullOrWhiteSpace(storageUri))
+            {
+                _testOutputHelper.WriteLine($"Using custom data lake storage uri {storageUri}");
+                _blobServiceClient = new BlobServiceClient(new Uri(storageUri), new DefaultAzureCredential());
+            }
+        }
+
+        [SkippableFact]
+        public async Task GivenValidDicomServer_WhenRun_CorrectResultShouldBeReturnedAsync()
+        {
+            Skip.If(_blobServiceClient == null);
+            await InitializeUniqueStorage();
+
+            IConfigurationRoot configuration = new ConfigurationBuilder()
+                .AddJsonFile(TestConfigurationPath)
+                .AddEnvironmentVariables()
+                .Build();
+
+            try
+            {
+                // Run e2e
+                using var tokenSource = new CancellationTokenSource();
+                IHost host = CreateHostBuilder(configuration).Build();
+                Task hostRunTask = host.RunAsync(tokenSource.Token);
+
+                CurrentTriggerEntity triggerEntity;
+                while (true)
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(10), CancellationToken.None);
+                    try
+                    {
+                        triggerEntity = await _metadataStore.GetCurrentTriggerEntityAsync(QueueTypeByte, CancellationToken.None);
+                        if (triggerEntity.TriggerStatus == TriggerStatus.Completed)
+                        {
+                            break;
+                        }
+                    }
+                    catch
+                    {
+                        // ignored
+                    }
+                }
+
+                tokenSource.Cancel();
+                await hostRunTask;
+
+                // check trigger
+                Assert.NotNull(triggerEntity);
+                Assert.Equal(TriggerStatus.Completed, triggerEntity.TriggerStatus);
+                Assert.Equal(0, triggerEntity.TriggerSequenceId);
+
+                long orchestratorJobId = triggerEntity.OrchestratorJobId;
+
+                // Check job status
+                JobInfo jobInfo = await _queueClient.GetJobByIdAsync(QueueTypeByte, orchestratorJobId, true, CancellationToken.None);
+                CheckJobStatus(jobInfo, "Dicom_All.json");
+
+                // Check result files
+                Assert.Equal(13, await GetResultFileCount(_blobContainerClient, "result/dicom"));
```

Comment:
You can use  `DicomConstants.DicomResourceType` here to avoid changes from "dicom" to "Dicom". I'll create another PR to update the names.

Reply:
Use the hard code in tests to ensure it works as expected.  I think we can modify it manually when changing from `dicom` to `Dicom`

Reply:
Sure. PR #246 is created to update naming.